https://issues.redhat.com/browse/ACM-13415 Unknown state troubleshoot (#6936)
open-cluster-management/rhacm-docs: clusters/support_troubleshooting/trouble_install_rosa_hcp.adoc

[#trouble-install-stuck-rosa-hcp]

> = Troubleshooting installation status stuck at `Installing` on a ROSA with HCP cluster

Reviewer: In 2.10, I did not see mentions of "HCP". HCP is also listed as a unapproved acronym. There is no short form approved based on the product names list.

> When installing {mce} on a Red Hat OpenShift Service on AWS (ROSA) with hosted control planes (HCP) cluster, {mce-short} becomes stuck at the `Installing` step and the `local-cluster` remains in the `Unknown` state.

Reviewer: ```suggestion
When you install {mce} on a Red Hat OpenShift Service on AWS (ROSA) with hosted control planes cluster, {mce-short} becomes stuck at the `Installing` step and the `local-cluster` remains in the `Unknown` state.
```

Reviewer: I also didn't see Red Hat OpenShift Service on AWS listed in the product list, but I do see that ROSA is used in the product docs here: https://docs.redhat.com/en/documentation/red_hat_openshift_service_on_aws/4/html/introduction_to_rosa/index

So it should be fine

[#symptom-install-stuck-rosa-hcp]
== Symptom: Installation status stuck at `Installing` on a ROSA with HCP cluster

> 10 minutes after installing {mce-short}, the `local-cluster` remains in the `Unknown` state. When you check the `klusterlet-agent` pod log in the `open-cluster-management-agent` namespace on your hub cluster, you see an error that resembles the following: 

Reviewer: It just seems a little odd to start with the number. What do you think about this suggestion?
```suggestion
The `local-cluster` remains in the `Unknown` state 10 minutes after installing {mce-short}. When you check the `klusterlet-agent` pod log in the `open-cluster-management-agent` namespace on your hub cluster, you see an error that resembles the following: 
```

> The `local-cluster` remains in the `Unknown` state 10 minutes after installing {mce-short}. When you check the `klusterlet-agent` pod log in the `open-cluster-management-agent` namespace on your hub cluster, you see an error that resembles the following: 

Reviewer: The following what? What do you think about this suggestion? 
```suggestion
The `local-cluster` remains in the `Unknown` state 10 minutes after installing {mce-short}. When you check the `klusterlet-agent` pod log in the `open-cluster-management-agent` namespace on your hub cluster, you see the following error message: 
```

[source,bash]
----
E0809 18:45:29.450874       1 reflector.go:147] k8s.io/client-go@v0.29.4/tools/cache/reflector.go:229: Failed to watch *v1.CertificateSigningRequest: failed to list *v1.CertificateSigningRequest: Get "https://api.xxx.openshiftapps.com:443/apis/certificates.k8s.io/v1/certificatesigningrequests?limit=500&resourceVersion=0": tls: failed to verify certificate: x509: certificate signed by unknown authority

> ----

Reviewer: I believe we agreed to not use `[source,bash]` for error messages and other messages that did not need to be copy. I'm not sure tho, I'm fine with leaving it as is

Author: Talked with Brandi about this recently, I was wrong and we actually need to add it to all terminal messages.

[#resolve-install-stuck-rosa-hcp]
== Resolving the problem: Installation status stuck at `Installing` on a ROSA with HCP cluster

To resolve the problem, add the `ISRG Root X1` certificate to the klusterlet CA bundle by completing the following steps:

. Download and encode the root CA certificate by running the following command:

+
[source,bash]
----
curl -s https://letsencrypt.org/certs/isrgrootx1.pem | base64 | tr -d "\n"
----

> . Create a `KlusterletConfig` resource add your encoded CA certificate to the `spec`. See the following example:

Reviewer: ```suggestion
. Create a `KlusterletConfig` resource by adding your encoded CA certificate to the `spec` parameter section. See the following example:
```

+
[source,yaml]
----
apiVersion: config.open-cluster-management.io/v1alpha1
kind: KlusterletConfig
metadata:
  name: isrg-root-x1-ca
spec:
  hubKubeAPIServerCABundle: "<your_ca_certificate>"
----

. Apply the resource by running the following command on the hub cluster:

+
[source,bash]
----
oc apply -f <filename>
----

. Annotate your managed cluster by running the following command on the hub cluster:

+
[source,bash]
----
oc annotate --overwrite managedcluster local-cluster agent.open-cluster-management.io/klusterlet-config='isrg-root-x1-ca'
----

. If the `local-cluster` state does not recover in 1 minute, export and decode the `import.yaml` file by running the following command on the hub cluster:

+
[source,bash]
----

> oc get secret local-cluster-import -n local-cluster -o jsonpath={.data.import .yaml} | base64 --decode > import.yaml

Reviewer 2: Use `jsonpath={.data.import\.yaml}` instead. The backslash `\` is needed to escape the dot `.` in the jsonpath.

Author: Updated

[#trouble-install-stuck-rosa-hcp]

> = Troubleshooting installation status stuck at `Installing` on a {rosa} hosted cluster

Reviewer 2: What's the official name of ROSA HCP? {rosa} hosted cluster or {rosa} with HCP cluster?

Author: @lahinson might know the answer here, can you chime in?

Reviewer 3: The official full name is a mouthful: "Red Hat OpenShift Service on AWS (ROSA) with hosted control planes (HCP)". The full name is used for the first mention. For subsequent mentions, you can use "ROSA with HCP".

Reviewer 3: In this case, since the name is in a heading, I'd use "ROSA with HCP", but use the full name on its first instance in the body text. So: "Troubleshooting installation status stuck at `Installing` on a ROSA with HCP cluster"

When installing {mce} on a {rosa} hosted cluster, {mce-short} becomes stuck at the `Installing` step and the `local-cluster` remains in the `Unknown` state.

[#symptom-install-stuck-rosa-hcp]
== Symptom: Installation status stuck at `Installing` on a {rosa} hosted cluster

> 10 minutes after installing {mce-short}, the `local-cluster` remains in the `Unknown` state. When you check the `klusterlet-agent` pod log in the `open-cluster-management-agent` namespace on your managed cluster, you see an error that resembles the following: 

Reviewer 2: on your managed cluster -> on your hub cluster

The managed cluster, in this case, is the hub cluster itself.

Author: Updated

[source,bash]
----
E0809 18:45:29.450874       1 reflector.go:147] k8s.io/client-go@v0.29.4/tools/cache/reflector.go:229: Failed to watch *v1.CertificateSigningRequest: failed to list *v1.CertificateSigningRequest: Get "https://api.xxx.openshiftapps.com:443/apis/certificates.k8s.io/v1/certificatesigningrequests?limit=500&resourceVersion=0": tls: failed to verify certificate: x509: certificate signed by unknown authority
----

[#resolve-install-stuck-rosa-hcp]
== Resolving the problem: Installation status stuck at `Installing` on a {rosa} hosted cluster

To resolve the problem, add the `ISRG Root X1` certificate to the klusterlet CA bundle by completing the following steps:

. Download and encode the root CA certificate by running the following command:

+
[source,bash]
----
curl -s https://letsencrypt.org/certs/isrgrootx1.pem | base64 | tr -d "\n"
----

. Create a `KlusterletConfig` resource add your encoded CA certificate to the `spec`. See the following example:

+
[source,yaml]
----
apiVersion: config.open-cluster-management.io/v1alpha1
kind: KlusterletConfig
metadata:
  name: isrg-root-x1-ca
spec:
  hubKubeAPIServerCABundle: "<your_ca_certificate>"
----

> . Apply the resource by running the following command:

Reviewer 2: . Apply the resource by running the following command on the hub cluster:

Author: Updated

+
[source,bash]
----
oc apply -f <filename>
----

> . Annotate your managed cluster by running the following command. Replace values where needed:

Reviewer 2: . Annotate your managed cluster by running the following command on the hub cluster:

Author: Updated

+
[source,bash]
----

> oc annotate --overwrite <cluster_name> local-cluster agent.open-cluster-management.io/klusterlet-config='isrg-root-x1-ca'

Reviewer 2: ```
oc annotate --overwrite managedcluster local-cluster agent.open-cluster-management.io/klusterlet-config='isrg-root-x1-ca'
```

Author: Updated

----

> . If the `local-cluster` state does not recover in 1 minute, export the `import.yaml` file by running the following command:

Reviewer 2: . If the `local-cluster` state does not recover in 1 minute, export the `import.yaml` file by running the following command on the hub cluster:

Author: Updated

+
[source,bash]
----
oc get secret local-cluster-import -n local-cluster -o jsonpath={.data.import .yaml} 
----

. Decode the file by running the following command

+
[source,bash]
----
base64 --decode > import.yaml

> ----

Reviewer 2: All those can be done by a singe command
```
oc get secret local-cluster-import -n local-cluster -o jsonpath='{.data.import\.yaml}' | base64 --decode > import.yaml
```

Author: We generally need to separate individual commands in the docs if they do different things

Reviewer 2: In my opinion, the commands are executed with a single goal – to export the import.yaml for a specific managed cluster. Therefore, it's more efficient to pipe them into a single command. However, if you still feel it's necessary to separate them, please refer to the content below:

---

5. If the local-cluster state does not recover in 1 minute, export the import.yaml file by running the following command on the hub cluster:
```
oc get secret local-cluster-import -n local-cluster -o jsonpath='{.data.import\.yaml}' > import-encoded.yaml
```
6.Decode the file by running the following command:
```
base64 --decode import-encoded.yaml > import.yaml
```
7....

Author: After discussing with the team we can make it a single command, thanks for the patience

Reviewer 4: We separate when we have, for example, two `oc` commands together but for two actions. We should make those into two steps. For this, I think it is very common to append a command with base64:

`Base64 is a generic term for a number of similar encoding schemes that encode binary data by treating it numerically and translating it into a base-64 representation. The Base64 term originates from a specific MIME-content transfer encoding.`

https://www.base64encode.org/

> . Apply the file by running the following command:

Reviewer 2: . Apply the file by running the following command on the hub cluster:

Author: Updated
